MTDUI/ModOptions.cs

```diff
@@ -25,27 +26,35 @@ public static void Register<T>(ConfigEntry<T> entry, List<T>? acceptableValues =
                 // add to acceptable values for enum
                 foreach (var value in enumValues) if (!acceptableValues.Contains((T)value)) acceptableValues.Add((T)value);
             }
-            else if(entry.SettingType == typeof(int) || entry.SettingType != typeof(float))
+            else if (entry.SettingType == typeof(int) || entry.SettingType != typeof(float))
             {
                 // really really need to do some better stuff here, but I don't care for now
                 // if they're not providing anything, all that should be provided is the default value. no changin!
-                if(!acceptableValues.Contains((T)entryBase.DefaultValue)) acceptableValues.Add((T)entryBase.DefaultValue);
+                if (!acceptableValues.Contains((T)entryBase.DefaultValue)) acceptableValues.Add((T)entryBase.DefaultValue);
             }
-            
-            if(entry.SettingType == typeof(bool))
+
+            if (entry.SettingType == typeof(bool))
             {
                 foreach (var value in new List<object> { false, true }) if (!acceptableValues.Contains((T)value)) acceptableValues.Add((T)value);
             }
 
-            // this is moderately insane and not how this should work
-            var asmName = Assembly.GetCallingAssembly().GetName().Name;
+            return acceptableValues.Cast<object>().ToList();
+        }
 
-            var modConfigEntry = new ModConfigEntry(entryBase, acceptableValues.Cast<object>().ToList());
+        public static void Register<T>(string subMenuName, ConfigEntry<T> entry, List<T>? acceptableValues = null, bool isInPause = false)
```

Comment:
`isInPause` would probably be better as an Enum, since it would make it explicitly clear what it does, plus add support for other configurations (Title Screen only, Pause Menu only, Both, possible future menus, etc)

I can always do this after merge though

Reply:
I pushing a new commit with this in a few minutes, and with the submenuname optional